umm-spec-lib/src/cmr/umm_spec/spatial_conversion.clj

```diff
@@ -1,12 +1,58 @@
 (ns cmr.umm-spec.spatial-conversion
   "Defines functions that convert umm spec spatial types to spatial lib spatial shapes."
   (:require
+   [clojure.string :as string]
    [cmr.spatial.line-string :as ls]
    [cmr.spatial.mbr :as mbr]
+   [cmr.common.xml.parse :as xml-parse]
    [cmr.spatial.point :as point]
    [cmr.spatial.polygon :as poly]
    [cmr.spatial.ring-relations :as rr]))
 
+(def valid-vertical-spatial-domain-types
+  "Valid values for VerticalSpatialDomainType according to UMM spec v1.10.0"
+  ["Atmosphere Layer"
+   "Maximum Altitude"
+   "Maximum Depth"
+   "Minimum Altitude"
+   "Minimum Depth"])
+
+(def squished-valid-vertical-spatial-domain-types
+  "Valid values for VerticalSpatialDomainType. Split and joined by \"\" to make
+   validation easier. DIF10 values are separated by `_` and all others are separated
+   by spaces."
+  ["AtmosphereLayer"
+   "MaximumAltitude"
+   "MaximumDepth"
+   "MinimumAltitude"
+   "MinimumDepth"])
+
+(defn vertical-spatial-domain-type-is-valid?
+  "Return true or false based on whether or not the given value matches
+   one of the valid values in valid-vertical-spatial-domain-types."
+  [vs-domain-type]
+  (let [squished-vs-domain-type (-> vs-domain-type
+                                    (string/split #"( |_)")
```

Comment:
Lines 35 and 36 can be replaced with `(string/replace #"( |_)" "")`

Reply:
But also, if MaximumAltitude is the original value, it'll say it's valid when it's not really valid, right?